File: 2021-8-12/2751.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

const num = parseInt(input[0]);
const numbers = input.slice(1).map(x => parseInt(x));


function quickSort(arr, start, end) {
    const part = partition(arr,start, end);

    if (start < part -1 ) {
        quickSort(arr,start,part -1); 
    }

    if (part < end ) {
        quickSort(arr,part,end);
    }

}

function partition(arr, start, end) {
    const pivot = arr[Math.floor((start+ end) /2)];
    while (start<= end) {
        while (arr[start]< pivot) {
            start++;
        }
        while (arr[end]> pivot) {
            end--;
        }

        if( start <= end ){
            swap(arr,start,end);
            start++;
            end--;
        }
    }
    return start;
}

function swap(arr,start,end) {
    const temp = arr[start];
    arr[start] = arr[end];
    arr[end] = temp;
}

// quickSort(numbers,0,numbers.length-1);
numbers.sort((a,b) => a-b);
console.log(numbers.join("\n"));

File: 2021-8-12/2751.py
import sys 
arr=[]
n=int(input())
for i in range(n):
    arr.append(int(sys.stdin.readline().rstrip()))
def QuickSort(left, right, arr):
    #left 가 클 경우 종료
    if(left>=right):
        return
    i = left
    j = right
    pivot = arr[(left+right)//2]
    #i, j가 교차 하면 종료

    while(i<=j):
        #i가 작은 거라면 통과 아니라면 멈춘다 -> 교환해야함
        while(arr[i] <pivot):
            i+=1
        #j가 큰 거라면 통과 아니라면 멈춘다 -> 교환해야함
        while(arr[j] >pivot):
            j-=1
        if(i<=j):
            temp=arr[i]
            arr[i]=arr[j]
            arr[j]=temp
            i+=1
            j-=1
        
    QuickSort(left,j,arr)
    QuickSort(i,right,arr)


QuickSort(0, n - 1, arr)
for i in range(n):
    print(arr[i])